packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/ClientMethodMapper.java

```diff
@@ -407,19 +407,31 @@ private List<ClientMethod> createClientMethods(Operation operation, boolean isPr
                     addClientMethodWithContext(methods, builder, parameters, getContextParameter(isProtocolMethod));
 
                     if (JavaSettings.getInstance().isSyncStackEnabled() && !proxyMethodUsesFluxByteBuffer) {
-                        // WithResponseAsync, with required and optional parameters
-                        methods.add(builder
+                        // WithResponseSync, with required and optional parameters
+                        builder
                             .returnValue(createSimpleSyncRestResponseReturnValue(operation,
                                 returnTypeHolder.syncReturnWithResponse, returnTypeHolder.syncReturnType))
                             .name(proxyMethod.getSimpleRestResponseMethodName())
                             .onlyRequiredParameters(false)
                             .type(ClientMethodType.SimpleSyncRestResponse)
                             .groupedParameterRequired(false)
-                            .methodVisibility(simpleSyncMethodVisibility)
-                            .proxyMethod(proxyMethod.toSync())
-                            .build());
+                            .proxyMethod(proxyMethod.toSync());
+                        if (settings.isFluent()) {
+                            // fluent + sync stack needs simple rest response for implementation only
```

Comment:
[generated code](https://github.com/azure-sdk/azure-sdk-for-java/blob/f3810211101a21375bcc6c990395cbc89ed393e0/sdk/computefleet/azure-resourcemanager-computefleet/src/main/java/com/azure/resourcemanager/computefleet/implementation/FleetsClientImpl.java#L632):
```java
public SyncPoller<PollResult<FleetInner>, FleetInner> beginCreateOrUpdate(String resourceGroupName,
    String fleetName, FleetInner resource) {
    Response<BinaryData> response = createOrUpdateWithResponse(resourceGroupName, fleetName, resource);
    return this.client.<FleetInner, FleetInner>getLroResult(response, FleetInner.class, FleetInner.class,
        Context.NONE);
}
```